src/hdmf/term_set.py

```diff
@@ -304,3 +299,85 @@ def extend(self, arg):
         else:
             msg = ('"%s" is not in the term set.' % ', '.join([str(item) for item in bad_data]))
             raise ValueError(msg)
+
+class TypeConfigurator:
+    """
+    This class allows users to toggle on/off a global configuration for defined data types.
+    When toggled on, every instance of a configuration file supported data type will be validated
+    according to the corresponding TermSet.
+    """
+    @docval({'name': 'path', 'type': str, 'doc': 'Path to the configuration file.', 'default': None})
+    def __init__(self, **kwargs):
+        self.config = None
+        if kwargs['path'] is None:
+            self.path = []
+        else:
+            self.path = [kwargs['path']]
+            self.load_type_config(config_path=self.path[0])
+
+    @docval({'name': 'data_type', 'type': str,
+             'doc': 'The desired data type within the configuration file.'},
+            {'name': 'namespace', 'type': str,
+             'doc': 'The namespace for the data type.'})
+    def get_config(self, data_type, namespace):
+        """
+        Return the config for that data type in the given namespace.
+        """
+        try:
+            namespace_config = self.config['namespaces'][namespace]
+        except KeyError:
+            msg = 'The namespace %s was not found within the configuration.' % namespace
+            raise ValueError(msg)
+
+        try:
+            type_config = namespace_config['data_types'][data_type]
+            return type_config
+        except KeyError:
+            msg = '%s was not found within the configuration for that namespace.' % data_type
+            raise ValueError(msg)
+
+    @docval({'name': 'config_path', 'type': str, 'doc': 'Path to the configuration file.'})
+    def load_type_config(self,config_path):
+        """
+        Load the configuration file for validation on the fields defined for the objects within the file.
+        """
+        try:
+            from ruamel.yaml import YAML
+        except ImportError: # pragma: no cover
+            msg = "Install yaml."
+            raise ValueError(msg)
```

Comment:
This should never happen because the import would fail way earlier. I think we can cut this out and move the import to the top of the file.

Reply:
Why would this never happen? If the user uses Termset without schemasheets, then they won't have this installed.

Reply:
`ruamel.yaml` is a requirement of HDMF: https://github.com/hdmf-dev/hdmf/blob/dev/pyproject.toml#L37
It's used in https://github.com/hdmf-dev/hdmf/blob/000020232cbf6dcfb053c4d57d984445c2eaa0e7/src/hdmf/spec/namespace.py#L2
and should be automatically imported when one runs `import hdmf`.

Reply:
Got it.